automata/base/automaton.py

```diff
@@ -22,10 +21,10 @@ class Automaton(metaclass=abc.ABCMeta):
     __slots__: Tuple[str, ...] = tuple()
 
     initial_state: AutomatonStateT
-    states: AbstractSet[AutomatonStateT]
```

Comment:
@eliotwrobson Is it okay to use Set instead of AbstractSet? They are equivalent and AbstractSet is defined as an alias for Set.

Reply:
I don't think this is true (see https://stackoverflow.com/a/50167430/2923069). The point if using `AbstractSet` is that both `Set` and `FrozenSet` can be used for this.

Reply:
@khoda81 I would prefer `AbstractSet`, since that is the convention @eliotwrobson and I have settled on for this project (given the reasons that Eliot linked to in his comment).